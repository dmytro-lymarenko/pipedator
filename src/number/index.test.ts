import { number } from './index';

describe('number()', () => {
	it('should validate a number', () => {
		expect(number().validate(1)).toBeNull();
		expect(number().validate(NaN)).toBeNull();
	});

	it('should return an error for non-number values', () => {
		expect(number().validate('5')).toMatchSnapshot();
		expect(number().validate({})).toMatchSnapshot();
		expect(number().validate([])).toMatchSnapshot();
		expect(number().validate(() => {})).toMatchSnapshot();
		expect(number().validate(null)).toMatchSnapshot();
		expect(number().validate(undefined)).toMatchSnapshot();
	});

	it('should support custom message', () => {
		expect(number('Custom message').validate('5')).toMatchSnapshot();
	});
});
