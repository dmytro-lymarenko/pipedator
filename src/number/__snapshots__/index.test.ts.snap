// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`number() should return an error for non-number values 1`] = `
Object {
  "children": null,
  "message": "Value should be a number",
  "path": "",
}
`;

exports[`number() should return an error for non-number values 2`] = `
Object {
  "children": null,
  "message": "Value should be a number",
  "path": "",
}
`;

exports[`number() should return an error for non-number values 3`] = `
Object {
  "children": null,
  "message": "Value should be a number",
  "path": "",
}
`;

exports[`number() should return an error for non-number values 4`] = `
Object {
  "children": null,
  "message": "Value should be a number",
  "path": "",
}
`;

exports[`number() should return an error for non-number values 5`] = `
Object {
  "children": null,
  "message": "Value should be a number",
  "path": "",
}
`;

exports[`number() should return an error for non-number values 6`] = `
Object {
  "children": null,
  "message": "Value should be a number",
  "path": "",
}
`;

exports[`number() should support custom message 1`] = `
Object {
  "children": null,
  "message": "Custom message",
  "path": "",
}
`;
